### PR TITLE
[feature-wip](duplicate_no_keys) add create duplicate table without keys

### LINKED ISSUE #18621: [Enhancement] duplicate table without key
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Doris skips data by short index, it is resource consuming to sorting data. Sometimes users do not need short index at all, so we can cancel enfore sort property on duplicate table.

### Solution

- ddl which creates a table without key
- load
- compaction
- query

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by nextdreamblue**
# Proposed changes

Issue Number: close #18621

## Problem summary

support create duplicate table without key columns
```
CREATE TABLE `test` (
  `user_id` bigint(20) NULL,
  `is_delete` tinyint(4) NULL,
  `client_version_int` int(11) NULL
) ENGINE=OLAP
COMMENT 'duplicate_no_keys'
DISTRIBUTED BY HASH(`user_id`) BUCKETS 4
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);
```
after this issue, user will get a duplicate table without key columns

add a fe conf `experimental_enable_duplicate_without_keys_by_default`, which whether create a duplicate table without keys by default when creating a table which not set key type and key columns.

if user want create a duplicate table without key, need set experimental_enable_duplicate_without_keys_by_default = true in fe.conf and create table like above sql.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p1

**Comment by nextdreamblue** [comment]
sh buildall

**Comment by nextdreamblue** [comment]
sh buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
sh buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 1]
```diff
@@ -2593,7 +2596,7 @@ private void createTablets(String clusterName, MaterializedIndex index, ReplicaS
     /*
      * generate and check columns' order and key's existence
      */
-    private void validateColumns(List<Column> columns) throws DdlException {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2596]: passing keysType may be a better way.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -3707,16 +3711,18 @@ public void setHaProtocol(HAProtocol protocol) {
         this.haProtocol = protocol;
     }
 
-    public static short calcShortKeyColumnCount(List<Column> columns, Map<String, String> properties)
-            throws DdlException {
+    public static short calcShortKeyColumnCount(List<Column> columns, Map<String, String> properties,
+                boolean isDupNoKeys) throws DdlException {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:3715]: passing keysType is a better way.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p1

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**